AmplifyPlugins/API/Tests/APIHostApp/AWSAPIPluginLazyLoadTests/LL12/DefaultPK/GraphQLLazyLoadDefaultPKTests.swift

```diff
@@ -101,6 +101,278 @@ final class GraphQLLazyLoadDefaultPKTests: GraphQLLazyLoadBaseTest {
                                                         where: DefaultPKChild.keys.id == defaultPKChild.id))
         assertList(queriedChildren, state: .isLoaded(count: 1))
     }
+
+    /*
+     - Given: Api category setup with DefaultPKModels
+     - When:
+        - Subscribe onCreate events of DefaultPKChild
+        - Create new DefaultPKChild instance with API
+     - Then:
+        - the newly created instance is successfully created through API. onCreate event is received.
+     */
+    func testSubscribeDefaultPKChildOnCreate() async throws {
+        await setup(withModels: DefaultPKModels())
+        let connected = asyncExpectation(description: "Subscription connected")
+        let onCreate = asyncExpectation(description: "onCreate received")
+        let child = DefaultPKChild()
+        let subscription = Amplify.API.subscribe(request: .subscription(of: DefaultPKChild.self, type: .onCreate))
+        Task {
+            do {
+                for try await subscriptionEvent in subscription {
```

Comment:
nit: Can we simplify the number of inner blocks for readability?

```swift
        Task {
            for try await subscriptionEvent in subscription {
                if subscriptionEvent.isConnected() {
                    await connected.fulfill()
                }
                if let newModel = subscriptionEvent.containsData(),
                   newModel.identifier == child.identifier {
                    await onCreate.fulfill()
                }
                if let error = subscriptionEvent.isError() {
                    XCTFail("Failed to create DefaultPKChild, error: \(error)")
                }
            }
        }

```

and have an extension:

```swift
extension GraphQLSubscriptionEvent {
    func isConnected() -> Bool {
        if case .connection(.connected) = self {
            return true
        }
        return false
    }

    func isError() -> Error? {
        if case .data(.failure(let error)) = self {
            return error
        }
        return nil
    }

    func containsData() -> T? {
        if case .data(.success(let data)) = self {
            return data
        }
        return nil
    }
}
```